.github/workflows/docker-release.yml

```diff
@@ -1,58 +1,102 @@
-      - name: ⚙️ Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
+
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@main
+
       - name: 🔐 Log in to Docker Hub
         uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
-      - name: 🛠️ Build & Push `juspay-dashboard-mcp` (Multi-Arch)
-        uses: docker/build-push-action@v5
-        with:
-          context: .
-          file: Dockerfile.dashboard
-          platforms: linux/amd64,linux/arm64
-          push: true
-          tags: |
-            ${{ secrets.DOCKERHUB_NAMESPACE }}/juspay-dashboard-mcp:latest
-            ${{ secrets.DOCKERHUB_NAMESPACE }}/juspay-dashboard-mcp:${{ github.ref_name }}
-      - name: 🛠️ Build & Push `juspay-mcp` (Multi-Arch)
-        uses: docker/build-push-action@v5
-        with:
-          context: .
-          file: Dockerfile
-          platforms: linux/amd64,linux/arm64
-          push: true
-          tags: |
-            ${{ secrets.DOCKERHUB_NAMESPACE }}/juspay-mcp:latest
-            ${{ secrets.DOCKERHUB_NAMESPACE }}/juspay-mcp:${{ github.ref_name }}
-      - name: 🛠️ Build & Push `juspay-dashboard-mcp for sse` (Multi-Arch)
-        uses: docker/build-push-action@v5
-        with:
-          context: .
-          file: Dockerfile.dashboard-sse
-          platforms: linux/amd64,linux/arm64
-          push: true
-          tags: |
-            ${{ secrets.DOCKERHUB_NAMESPACE }}/juspay-dashboard-mcp:sse-latest
-            ${{ secrets.DOCKERHUB_NAMESPACE }}/juspay-dashboard-mcp:sse-${{ github.ref_name }}
-      - name: 🛠️ Build & Push `juspay-mcp for sse` (Multi-Arch)
-        uses: docker/build-push-action@v5
+
+      - name: Build and push all images
+        run: |
+          # Function to build, tag and push images for releases
+          build_and_push_release() {
+            local nix_target="$1"
+            local source_tag="$2"
+            local target_image="$3"
+            local tag_prefix="$4"
+            local version="${{ github.ref_name }}"
+            local arch_suffix="${{ matrix.tag_suffix }}"
+
+            local final_tag="${tag_prefix}${version}-${arch_suffix}"
+
+            echo "Building and pushing $source_tag -> $target_image:$final_tag"
+            nix run .#${nix_target}.copyToDockerDaemon --system ${{ matrix.system }}
+            docker tag ${source_tag}:latest ${target_image}:${final_tag}
+            docker push ${target_image}:${final_tag}
+          }
+
+          # Build and push all images
+          build_and_push_release "docker" "juspay-mcp" "$IMAGE_MCP" ""
+          build_and_push_release "docker-sse" "juspay-mcp-sse" "$IMAGE_MCP" "sse-"
+          build_and_push_release "docker-dashboard" "juspay-dashboard-mcp" "$IMAGE_DASHBOARD" ""
+          build_and_push_release "docker-dashboard-sse" "juspay-dashboard-mcp-sse" "$IMAGE_DASHBOARD" "sse-"
```

Comment:
Having separate CI steps for these would be nice (since it makes CI logs more legible when something fails). Same for `create_release_manifest` below. 

I suppose putting those bash functions in a separate file and sourcing it would facilitate that?

Reply:
Example, https://github.com/juspay/haskell-incremental-build-template/tree/61403f14dde48995d86e48b1e7f3902b63bbad1a/.github/workflows